capstone/src/main/java/com/google/univiz/api/MapData.java

```diff
@@ -1 +1,13 @@
 package com.google.univiz.api;
+
+/*
+ * The MapData interface is responsible for all data management/processing specific for the maps 
+ * feature of the UniViz webapp.
+ */
+public interface MapData {
+  /*
+   * Takes a list of college names and returns a CollegeMapsData object to the caller. This object 
+   * contains all data relevant to the Maps feature.
+   */
+  CollegeMapsData getMapData(List<String> collegeNames);
```

Comment:
Why `CollegeMapsData` instead of `MapsData`?

Reply:
MapsData is the name of the interface so I didn't want to use the same name for the possible object name...

Reply:
I think the general practice is to use the interface for the return type if possible to achieve maximal abstraction.

Reply:
Hi Ihsan, I didn't quite understand what you meant by using the interface as the return type? Can you expand on that? Thank you!

Reply:
Ihsan's suggestion is not to use the exact interface name for return type (MapsData != MapData). I think he's saying the return type name should be similar (driven from) the interface name? For example the `MapData` api returns `MapsData` objects. I personally don't have a strong opinion.

Ihsan, remember that generally the burden of proof is on the code reviewer. If you're making a suggestion, you also need to provide the reasoning (preferably with citation) why you think this is the case.

Reply:
That is true, I could have provided better evidence of what I was talking about.

I actually mistyped and meant `MapData`. And I also thought `CollegeMapsData` will be implementing `MapData`, so my thought process was `CollegeMapsData` should only be the return type if it offers something additional that returning `MapData` would not. Similar to how `List.subList` does not return `ArrayList` even though `ArrayList` would suffice in multiple scenarios.

I should have also mentioned it's more of a stylistic thing that isn't significant enough to block the PR on, but more of something for all of us to think about.